src/ir/effects.h

```diff
@@ -66,10 +66,16 @@ class EffectAnalyzer {
   // Walk an entire function body. This will ignore effects that are not
   // noticeable from the perspective of the caller, that is, effects that are
   // only noticeable during the call, but "vanish" when the call stack is
-  // unwound.
+  // unwound. Unlike walking just the body, walking the function will also
+  // include the effects of any return calls the function makes.
```

Comment:
```suggestion
  // unwound.
  //
  // Unlike walking just the body, walking the function will also
```suggestion
  // include the effects of any return calls the function makes. For that
  // reason, it is a bug if a user of this code calls walk(Expression* and not
  // walk(Function*) if their intention is to scan an entire function body.
  // Putting it another way, a return_call is syntax sugar for a return and a
  // call, where the call executes at the function scope, so there is a
  // meaningful difference between scanning an expression and scanning
  // the entire function body.
```